stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -321,19 +322,20 @@ class cholesky_opencl : public vari {
    *
    */
   virtual void chain() {
-    auto L_adj_cpu = Eigen::MatrixXd::Zero(M_, M_).eval();
-    auto L_cpu = Eigen::MatrixXd::Zero(M_, M_).eval();
+    int packed_size = M_ * (M_ + 1) / 2;
+    std::vector<double> L_adj_cpu;
+    L_adj_cpu.reserve(packed_size);
+    std::vector<double> L_val_cpu;
+    L_val_cpu.reserve(packed_size);
```

Comment:
The power of benchmarking:
```
---------------------------------------------------------------
Benchmark                     Time             CPU   Iterations
---------------------------------------------------------------
push                       5100 ns         4505 ns       135015
reserve_push               2474 ns         2371 ns       297215
initialize_op_assign        900 ns          892 ns       652486
```
Using this benchmark:
https://github.com/seantalts/perf-math/blob/master/stdvectorfun.cpp

I just whipped that up and it's late here (I'm not in EST, I'm actually in Portugal!) but seems like we should just do the naive thing 😂 

tagging @SteveBronder and honestly if this result holds up it seems like we should publish unintuitive stuff like this somewhere other Stan C++ devs will read...